internal/webhook/v1/alertmanager_config_secret_webhook.go

```diff
@@ -19,79 +19,149 @@ package v1
 //
 // NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
 // as this struct is used only for temporary operations and does not need to be deeply copied.
-type SecretCustomValidator struct {
-	// TODO(user): Add more fields as needed for validation
+type AlertmanagerConfigSecretCustomValidator struct {
+	client client.Client
 }
 
-var _ webhook.CustomValidator = &SecretCustomValidator{}
+var _ webhook.CustomValidator = &AlertmanagerConfigSecretCustomValidator{}
 
 // ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type Secret.
-func (v *SecretCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+func (v *AlertmanagerConfigSecretCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
 	secret, ok := obj.(*corev1.Secret)
 	if !ok {
 		return nil, fmt.Errorf("expected a Secret object but got %T", obj)
 	}
-	secretlog.Info("Validation for Secret upon creation", "name", secret.GetName())
+	log.Info("Validation for Secret upon creation", "name", secret.GetName())
 
-	// TODO(user): fill in your validation logic upon object creation.
+	if err := v.validateConfiguredTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	if err := v.validateNoDuplicateTenant(ctx, secret); err != nil {
+		return nil, err
+	}
+	return nil, validateAlertmanagerConfig(ctx, secret)
+}
 
-	return nil, nil
+func (v *AlertmanagerConfigSecretCustomValidator) validateConfiguredTenant(ctx context.Context, secret *corev1.Secret) error {
+	tenants, err := tenancy.ListTenants(ctx, v.client)
+	if err != nil {
+		return fmt.Errorf("failed to list tenants: %w", err)
+	}
+
+	if tenant, ok := secret.Labels[tenancy.TenantSelectorLabel]; !ok {
+		return fmt.Errorf("tenant label is required")
+	} else if !slices.Contains(tenants, tenant) {
+		return fmt.Errorf("tenant %s is not in the list of accepted tenants", tenant)
+	}
+	return nil
+}
+
+// validateNoDuplicateTenant ensures that no other secret with the same tenant label exists.
+func (v *AlertmanagerConfigSecretCustomValidator) validateNoDuplicateTenant(ctx context.Context, secret *corev1.Secret) error {
+	tenant, ok := secret.Labels[tenancy.TenantSelectorLabel]
+	if !ok {
+		return fmt.Errorf("tenant label is required")
+	}
+
+	var secretList corev1.SecretList
+	if err := v.client.List(ctx, &secretList, client.InNamespace(""),
+		client.MatchingLabelsSelector{
+			Selector: labels.SelectorFromSet(
+				labels.Set{
+					predicates.AlertmanagerConfigSelectorLabelName: predicates.AlertmanagerConfigSelectorLabelValue,
+					tenancy.TenantSelectorLabel:                    tenant,
+				},
+			),
+		}); err != nil {
+		return fmt.Errorf("failed to list secrets for tenant %s: %w", tenant, err)
+	}
+
+	if len(secretList.Items) > 0 && (secretList.Items[0].Name != secret.Name ||
+		secretList.Items[0].Namespace != secret.Namespace) {
```

Comment:
Should check for all items in the list here, I get that having more than 1 alertmanager config per tenant should not happen but I'd rather be safe here.